src/sagas/activity/sagas.js

```diff
@@ -0,0 +1,61 @@
+import { put, take, call } from 'redux-saga/effects';
+
+import { messagesAdded, AddNewMessageAction } from '../../actions/activity';
+import defaultDB from '../../services/database';
+import {
+  convertFromDatabase,
+  convertToDatabase,
+} from '../../utils/mapping/activity';
+import type { ActivityLogMessage } from '../../types/ActivityLogMessage';
+import { createDatabaseUpdateChannel } from '../database';
+
+
+/**
+ * @desc Function that listen to the new message addition event
+ * @return {void}
+ */
+export function* watchNewMessages(): Generator<*, *, *> {
+  const db = yield defaultDB;
+  const results = db.objects('MessageJob');
+  const channel = createDatabaseUpdateChannel(results);
+  while (true) {
+    const { collection } = yield take(channel);
+    yield put(messagesAdded(collection.map(convertFromDatabase)));
+  }
+}
+
+const buildMessageObject = (
+  highestId,
+  message,
+  params,
+  interpret,
+): ActivityLogMessage => ({
+  id: highestId,
+  msg: message,
+  params: JSON.stringify(params),
+  interpret,
+  created_at: new Date(),
+});
+
+/**
+ * @desc Add a new activity log into the database
+ * @param {AddNewMessageAction} action ADD_MESSAGE action
+ * @return {void}
+ */
+export function* addNewMessageSaga(action: AddNewMessageAction) {
+  const db = yield defaultDB;
+  const params = action.params || {};
+  const interpret = action.interpret !== false;
+  const messages = db.objects('MessageJob').sorted('id', false);
```

Comment:
Did you add test's for the database insertion? sometimes realm doesn't work as expected so it's good to be on the safe side ...

Reply:
I did not add tests for database functions yet.
This logic has not been completed yet and we have additional requirements which we will handle in another ticket.
I can add tests in the next phase because we have to modify the logic significantly.